students/avalvillar/lesson4/mailroom_update.py

```diff
@@ -0,0 +1,153 @@
+"""
+    Author is Antonio V. Alvillar
+    Self-Paced-Online Python 210 UWPCE
+    December 11th 2018
+"""
+
+"""
+This function has been updated to use a dictionary and fileout.
+    There are quite a bit of prints that were used during testing. This
+    can be deleted if need be. Keeping them there in case of bug fixes
+    needed with this assignment. With the list of lists I felt it important.
+"""
+
+#Initial Donor list with donations and then print for checking.
+donors = {'Adam Alvarez': [1.25], 'Beth Bonnor':[2.73]}
+donors['Cate Campbell'] = [3.33, 4.44]
+donors['Dave Don'] = [5550.33, 1.41]
+donors['Eric Ebron'] = [12.02, 5.25, 4.25]
+#print(donors)
+
+
+#Display the Thank you note to the screen
+def display_thank_you(donor, amount): #Create an email with name an donation.
+    personal_info = {'name':donor, 'donation':amount}
+    print('\n')
+    print("{name}, \n\n    Your donation of ${donation}".format(**personal_info)
+          + " was received and we are forever grateful for your support." +
+          "\n\nSincerely,\nDonation Center\n(555) 555-5555")
+    #print("\n\n" + email)
+    print("\n\n\nReturning to the main menu.....")
+    start_up() #After email is displayed - start over.
+
+#Send a Thank You note function
+def thank_you():
+    print("\n(You can type \"list\" to display current donors)")
+    name = input("What is the First and Last name of the recipient? ")
+    #print(name) #Display the name input for testing (before checks)
+    if name.lower() == 'q': #Return to start if q is typed
+        start_up()
+    while name.lower() == 'list': #loop until a name other than list is given.
+        print()
+        print('[ ', end='')
+        for key in donors:
+            print(key, end=', ')
+        print(']') #New line after list is printed.
+        print()
+        name = input("Type a full name from the list or provide a new one? ")
+        if name.lower() == 'q': #Return to start if q is typed
+            start_up()
+    donation = input("How much is the donation? ")
+    if donation.lower() == 'q':#Return to start if q is typed
+        start_up()
+    found = False #Trigger to see if a name is found in current list
+    for item in donors:
+        if name in item:
+            donors[name].append(float(donation))
+            #print("*Existing Donor Found: Donation Added*")
+            #print(donors) #Verification that a donation was added to donor.
+            found = True #Donor was found - do not add.
+    if not found: #If no donor was found - add to list.
+        donors[name] = [float(donation)]
+        #print("*New Donor Added*")
+        #print(donors) #Verification that a new donor and donation was added.
```

Comment:
you don't need to loop over a `dict`, which is why it's efficient in this use case:
```
if donors.get(name):
    # key is found
    donors[name].append(donation)
else:
    # key not found
    donors[name] = [donation]
# done!
```